src/Datadog.Trace/Tracer.cs

```diff
@@ -55,7 +55,7 @@ internal Tracer(TracerSettings settings, IAgentWriter agentWriter, ISampler samp
             Settings = settings ?? TracerSettings.FromDefaultSources();
             _agentWriter = agentWriter ?? new AgentWriter(new Api(Settings.AgentUri));
             _scopeManager = scopeManager ?? new AsyncLocalScopeManager();
-            Sampler = sampler ?? new RateByServiceSampler();
+            Sampler = sampler ?? new RuleBasedSampler(new RateLimiter(Settings.MaxTracesSubmittedPerSecond));
```

Comment:
I like the composition here 👍

Reply:
is this changing the default to the new rule based sampler?

Reply:
No, old logic is within this sampler.